File: python/atelier_core/models/signature.py
"""Signature captured on collection."""

from __future__ import annotations


class Signature:
    def __init__(self, id: int, repair_order_id: int, signed_by: str, svg_path: str = "") -> None:
        self.id = id
        self.repair_order_id = repair_order_id
        self.signed_by = signed_by
        self.svg_path = svg_path

    def is_captured(self) -> bool:
        return self.svg_path != ""

File: go/core/models/signature.go
package models

// Signature is the collection signature on a completed repair.
type Signature struct {
	ID            int
	RepairOrderID int
	SignedBy      string
	Witnessed     bool
}

// NewSignature builds an unwitnessed signature.
func NewSignature(id int, repairOrderID int, signedBy string) *Signature {
	return &Signature{ID: id, RepairOrderID: repairOrderID, SignedBy: signedBy}
}

// IsValid is true once somebody signed.
func (s *Signature) IsValid() bool {
	return s.SignedBy != ""
}
